src/snowcli/cli/nativeapp/manager.py

```diff
@@ -270,8 +274,22 @@ def _apply_package_scripts(self) -> None:
                 raise InvalidPackageScriptError(relpath, e)
 
         # once we're sure all the templates expanded correctly, execute all of them
-        if self.warehouse:
-            self._execute_query(f"use warehouse {self.warehouse}")
+        try:
+            if self.package_warehouse:
+                self._execute_query(f"use warehouse {self.package_warehouse}")
+            elif self._conn.warehouse:
+                self._execute_query(f"use warehouse {self._conn.warehouse}")
+            else:
+                raise MissingWarehouseError(
+                    f"""
+                    No warehouse specified for provider role to use.
+                    Please add a warehouse for your provider role in your config.toml file, or
+                    project definition file or via command line.
+                    """
+                )
+        except Exception as err:
```

Comment:
For those reviewing: 
The reason why we raise an error but also catch it and only `log.info` is it accommodate the case where a provider can argue they do not want to specify a warehouse because none of their package scripts require use of warehouse. In that case, we do not want to force them to have something here.
But at least printing this will allow for debugging for providers that should have provided a warehouse.